src/engine/qcommon/files.cpp

```diff
@@ -627,12 +627,18 @@ bool FS_LoadServerPaks(const char* paks, bool isDemo)
 	for (auto& x: args) {
 		std::string name, version;
 		Util::optional<uint32_t> checksum;
-		if (!FS::ParsePakName(x.data(), x.data() + x.size(), name, version, checksum)) {
-			Com_Error(errorParm_t::ERR_DROP, "Invalid pak reference from server: %s", x.c_str());
-		} else if (!checksum) {
-			if (isDemo || allowRemotePakDir.Get())
-				continue;
-			Com_Error(errorParm_t::ERR_DROP, "The server is configured to load game data from a directory which makes it incompatible with remote clients.");
+		if (Str::IsSuffix(FS::LEGACY_PAK_ZIP_EXT, x.data())) {
```

Comment:
I refactored that part, I was also able to remove the `AddLegacyPak` function I added before… :smiley: